src/language-handlebars/utils.js

```diff
@@ -11,6 +11,84 @@ function isGlimmerComponent(node) {
   return tagFirstChar.toUpperCase() === tagFirstChar || isLocal;
 }
 
+function isWhitespaceNode(node) {
+  return node.type === "TextNode" && !/\S/.test(node.chars);
+}
+
+function isNodeOfSomeType(node, types) {
+  return node && types.some(type => node.type === type);
+}
+
+function isParentOfSomeType(path, types) {
+  const parentNode = path.getParentNode(0);
+  return isNodeOfSomeType(parentNode, types);
+}
+
+function isPreviousNodeOfSomeType(path, types) {
+  const previousNode = getPreviousNode(path);
+  return isNodeOfSomeType(previousNode, types);
+}
+
+function isNextNodeOfSomeType(path, types) {
+  const nextNode = getNextNode(path);
+  return isNodeOfSomeType(nextNode, types);
+}
+
+function getPreviousNode(path, lookBack = 1) {
+  const node = path.getValue();
+  const parentNode = path.getParentNode(0);
+
+  const children = parentNode && (parentNode.children || parentNode.body);
+  if (children) {
+    const nodeIndex = children.indexOf(node);
+    if (nodeIndex > 0) {
+      const previousNode = children[nodeIndex - lookBack];
+      return previousNode;
+    }
+  }
+}
+
+function getNextNode(path) {
+  const node = path.getValue();
+  const parentNode = path.getParentNode(0);
+
+  const children = parentNode.children || parentNode.body;
+  if (children) {
+    const nodeIndex = children.indexOf(node);
+    if (nodeIndex < children.length) {
+      const nextNode = children[nodeIndex + 1];
+      return nextNode;
+    }
+  }
+}
```

Comment:
```suggestion
function getNextNode(path) {
  const node = path.getValue();
  const parentNode = path.getParentNode(0);
  const children = parentNode.children || parentNode.body;
  return children[children.indexOf(node) + 1];
}
```

Reply:
We are not check `index === -1`, I don't think need check `children` either

Reply:
My initial goal was more to extract these utils than to refactor them. But there is definitely work to do in this area. 😄

Reply:
Nice one

Reply:
Should be safe

Reply:
`getPreviousNode` checked `nodeIndex > 0` , I think we should also check it in `getPreviousNode`

Reply:
I updated the suggestion

Reply:
How about ...

```js
const getNextNode = node => getPreviousNode(node, -1);
```

Reply:
> getPreviousNode checked nodeIndex > 0 , I think we should also check it in getNextNode, that one also checked parentNode

agreed